contributors/devel/sig-testing/verify-tests.md

```diff
@@ -105,3 +105,81 @@ Finally the script checks that all external group versions
 (e.g. `foobar/v1`) are defined in `hack/lib/init.sh` in either the
 `KUBE_AVAILABLE_GROUP_VERSIONS` or `KUBE_NONSERVER_GROUP_VERSIONS`
 bash variables.
+
+### `verify-bazel`
+
+This verification script validates whehter the removal of `bazel`
+related files is still needed or not. This script after setting the
+temporary directories for gopath and kube-related path, runs
+`[update-bazel.sh]()`. And once executed, `update-bazel` deletes all
+the bazel related files which are either not there in vendor and some
+additional one-off files. After execution, the `verify-bazel` script
+again calculates diff to check the earlier output. And if the diff
+result is non-zero then script `update-bazel.sh` is run again for the
+files that might be left.  
+
+Following are the files that are searched for deletion as part of this
```

Comment:
this seems excessive to document here and likely to become stale.

IMHO we should document the intent of the tools, not the implementation details which will get out of sync with the source.

Reply:
@BenTheElder so do you mean, we need to document something like "why we need a particular `verify-` script" rather than "what this script does?". If so, do you have some references which I can look at and will make the changes accordingly 🙂

Reply:
I mean the snippet starting here that lists *exactly* what it finds is excessive. We should document it at a high level, not e.g. every file path it looks at, which is subject to change and will not stay in sync.

Reply:
It is also not important what all files it looks at for the user to use it.

Reply:
We should document the high-level intent, the implementation is open source if a developer wishes to see *exactly* what it does step by step.

Reply:
Sure @BenTheElder , thanks for suggesting the correct way 🙂